Mathlib/Topology/Algebra/Group/Profinite.lean

```diff
@@ -0,0 +1,262 @@
+
+lemma Subgroup.discreteTopology  (U_open : IsOpen (U : Set G)) : DiscreteTopology (G ⧸ U) := by
+  apply singletons_open_iff_discrete.mp
+  rintro ⟨g⟩
+  erw [isOpen_mk, QuotientGroup.preimage_mk_singleton_mk]
+  exact Homeomorph.mulLeft g |>.isOpen_image|>.mpr U_open
+
+lemma Subgroup.finiteQuotient [CompactSpace G] (U_open : IsOpen (U : Set G)) : Finite (G ⧸ U) :=
+  have : CompactSpace (G ⧸ U) := Quotient.compactSpace
+  have : DiscreteTopology (G ⧸ U) := U.discreteTopology U_open
+  finite_of_compact_of_discrete
+
+end CompactQuotient
+
+section ConjAction
+
+open ConjAct
+
+instance : MulAction (ConjAct G) (Subgroup G) := Subgroup.pointwiseMulAction
+
+lemma intersectionOfConjugatesIsNormal (H : Subgroup G) :
+    (⨅ (g : ConjAct G), g • H).Normal := by
+  constructor
+  intro n hn g
+  simp only [Subgroup.mem_iInf] at *
+  intro k
+  obtain ⟨h, h_in_H : h ∈ H, hh : ((toConjAct g)⁻¹ * k) • h = n⟩ := hn _
+  exists h
+  simp only [MulDistribMulAction.toMonoidEnd_apply, MulDistribMulAction.toMonoidHom_apply]
+  rw [←hh, smul_def, smul_def]
+  simp only [map_mul, map_inv, ofConjAct_toConjAct, mul_inv_rev, inv_inv]
+  group
+  exact And.intro h_in_H trivial
+
+lemma conjActionOrbitSubgroup_def (H : Subgroup G) :
+    ⨅ (g : ConjAct G), g • H = ⨅ C ∈ MulAction.orbit (ConjAct G) H, C := by
+  ext
+  simp [Subgroup.pointwise_smul_def, MulAction.mem_orbit_iff, Subgroup.mem_iInf]
+
+lemma stabilizer_eq_normalizer (H : Subgroup G) :
+    MulAction.stabilizer (ConjAct G) H = H.normalizer := by
+  ext
+  erw [MulAction.mem_stabilizer_iff, ←eq_inv_smul_iff, Subgroup.mem_normalizer_iff, SetLike.ext_iff]
+  simp only [Subgroup.mem_inv_pointwise_smul_iff]
+  rfl
+
+lemma subgroup_stabilizes_itself (H : Subgroup G) : H ≤ MulAction.stabilizer (ConjAct G) H := by
+  rw [stabilizer_eq_normalizer]
+  exact Subgroup.le_normalizer
+
+lemma QuotientGroup.finite_of_le {H K : Subgroup G} (le : H ≤ K) (fh : Finite (G ⧸ H)) :
+    Finite (G ⧸ K) := by
+  apply Nat.finite_of_card_ne_zero
+  intro (n_0 : K.index = 0)
+  have := Subgroup.index_dvd_of_le le
+  simp_all only [zero_dvd_iff, Subgroup.index_ne_zero_of_finite]
+
+lemma Subgroup.finiteConjugationOrbit_of_finiteIndex (H : Subgroup G) (hf : H.index ≠ 0) :
+    Finite (MulAction.orbit (ConjAct G) H) := by
+  rw [Equiv.finite_iff <| MulAction.orbitEquivQuotientStabilizer (ConjAct G) H]
+  exact QuotientGroup.finite_of_le (subgroup_stabilizes_itself H) (Nat.finite_of_card_ne_zero hf)
+
+end ConjAction
+
+section ContConjAction
+
+variable [TopologicalSpace G] [TopologicalGroup G]
+
+open ConjAct
+
+lemma conjOpen_ofOpen (U : Subgroup G) (U_open : IsOpen (U : Set G)) (g : ConjAct G) :
+    IsOpen (g • U : Set G) := by
+  apply (Homeomorph.isOpen_image
+    (Homeomorph.trans (Homeomorph.mulLeft (ofConjAct g)) (Homeomorph.mulRight (ofConjAct g)⁻¹))).mpr
+  assumption
+
+end ContConjAction
+
+section TopGroups
+
+def subGroup_in_set (U : Set G) (one_mem : 1 ∈ U) : Subgroup G :=
```

Comment:
I added this auxiliary definition to split the proof of `clopenContainsOpenSubgroup` , but I suppose this definition has no other use. Is this still preferred over defining this in the body of `clopenContainsOpenSubgroup`? Can I somehow mark this as "private", i.e. should not be imported when importing the file?